py3status/modules/weather_owm.py

```diff
@@ -0,0 +1,751 @@
+a few groups, and due to the limitations of UTF-8, this is really as expressive
+as it gets.
+
+I would highly suggest you install an additional font, such as the incredible
+(and free!) Weather Icons font (https://erikflowers.github.io/weather-icons),
+which has icons for most weather scenarios. But, this will still work with
+the i3bar default font, Deja Vu Sans Mono font, which has Unicode support.
+You can see the (limited) weather icon support within Unicode in the defaults.
+
+For more information, see the documentation
+(https://openweathermap.org/weather-conditions) on what weather conditions are
+supported. See the configuration options for how to specify each weather icon.
+
+Configuration parameters:
+    api_key: Your OpenWeatherMap API key
+        See https://openweathermap.org/appid. Required!
+        (default None)
+    cache_timeout: The time between weather polling in seconds
+        It is recommended to keep this at a higher value to avoid rate
+        limiting with the API's.
+        (default 600)
+    forecast_days: Number of days to include in the forecast, including today
+        (regardless of the 'forecast_include_today' flag)
+        (default 3)
+    forecast_include_today: Include today in the forecast? (Boolean)
+        (default False)
+    forecast_text_separator: Separator between entries in the forecast
+        (default ' ')
+    format: How to display the weather
+        This also dictates the type of forecast. The placeholders here refer to
+        the format_[...] variables found below.
+        Available placeholders:
+            icon, city, clouds, rain, snow, wind, humidity, pressure,
+            temperature, sunrise, sunset, desc, desc_long, forecast
+        (default '{city}: {icon} {temperature} {desc} {forecast}')
+    format_clouds: Formatting for cloud coverage (percentage)
+        Available placeholders:
+            icon, coverage
+        (default '{icon} {coverage}%')
+    format_forecast: Formatting for future forecasts
+        Available placeholders:
+            See 'format'
+        This is similar to the 'format' field, but contains information
+        for future weather. Notably, this does not include information about
+        sunrise or sunset times.
+        (default '{icon}')
+    format_humidity: Formatting for humidity (percentage)
+        Available placeholders:
+            icon, humidity
+        (default '{icon} {humidity}%')
+    format_pressure: Formatting for atmospheric pressure
+        Available placeholders:
+            icon, pressure, sea_level
+        (default '{icon} {pressure}hPa')
+    format_rain: Formatting for rain volume over the past 3 hours
+        Available placeholders:
+            icon, amount
+        (default '{icon} {amount:.0f}in')
+    format_snow: Formatting for snow volume over the past 3 hours
+        Available placeholders:
+            icon, amount
+        (default '{icon} {amount:.0f}in')
```

Comment:
seeing how we know the unit `snow_unit` could we not us it here eg `'{icon} {amount:.0f}{snow_unit}'` so that when we change the unit the display updates too.

I'm not sure we need `snow_unit` and `rain_unit` etc maybe `unit_depth` or something but maybe the extra flexibility is worth it.

I still need to try this module for real but I think the idea behind it is great